discovery/v1.js

```diff
@@ -0,0 +1,331 @@
+      json: true
+    },
+    requiredParams: ['environment_id'],
+    defaultOptions: this._options
+  };
+  return requestFactory(parameters, callback);
+};
+
+/**
+ * Get details about a collection
+ *
+ * @param {Object} params
+ * @param {String} params.environment_id
+ * @param {string} params.collection_id
+ */
+DiscoveryV1.prototype.getCollection = function(params, callback) {
+  params = params || {};
+
+  var parameters = {
+    options: {
+      url: '/v1/environments/{environment_id}/collections/{collection_id}',
+      method: 'GET',
+      path: pick(params, ['environment_id', 'collection_id']),
+      json: true
+    },
+    requiredParams: ['environment_id', 'collection_id'],
+    defaultOptions: this._options
+  };
+  return requestFactory(parameters, callback);
+};
+
+/**
+ * Create a new collection
+ *
+ * @param {Object} params
+ * @param {String} params.environment_id environment guid for the collection
+ * @param {string} params.collection_name
+ * @param {string} params.description
+ * @param {string} params.configuration_id  configuration to create the collection in
+ * @param {string} params.language_code currently, only `en_us` is supported
+ */
+DiscoveryV1.prototype.createCollection = function(params, callback) {
+  params = params || {};
+
+  params.language_code = params.language_code || 'en_us';
+
+  var parameters = {
+    options: {
+      url: '/v1/environments/{environment_id}/collections',
+      method: 'POST',
+      path: pick(params, ['environment_id']),
+      multipart: [
+        {
+          'content-type': 'application/json',
+          body: JSON.stringify(pick(params,['collection_name','description','configuration_id','language_code']))
+        }
+      ],
+      json: true
+    },
+    requiredParams: ['environment_id', 'configuration_id', 'collection_name'],
+    defaultOptions: this._options
+  };
+  return requestFactory(parameters, callback);
+};
+
+/**
+ * Delete a collection
+ *
+ * @param {Object} params
+ * @param {String} params.environment_id environment guid for the collection
+ * @param {string} params.collection_id the guid of the collection to delete
+ */
+DiscoveryV1.prototype.deleteCollection = function(params, callback) {
+  params = params || {};
+
+  params.language_code = params.language_code || 'en_us';
+
+  var parameters = {
+    options: {
+      url: '/v1/environments/{environment_id}/collections/{collection_id}',
+      method: 'DELETE',
+      path: pick(params, ['environment_id', 'collection_id']),
+      json: true
+    },
+    requiredParams: ['environment_id', 'collection_id'],
+    defaultOptions: this._options
+  };
+  return requestFactory(parameters, callback);
+};
+
+/**
+ * Add a document to a collection
+ * @param params
+ * @param {String} params.environment_id environment guid for the collection
+ * @param {string} params.collection_id the guid of the collection to delete
+ * @param {file} params.file a file to post (smaller than 50mb)
```

Comment:
I think this should be `{Buffer|ReadableStream|Object}` rather than `{file}` (?)

(Object allows you to specify filename and/or content type, e.g. `{value: Buffer, options: contentType: 'text/foo'}`.. or something like that.)